search/facets_builder.go

```diff
@@ -321,7 +321,11 @@ func (fr *FacetResult) Merge(other *FacetResult) {
 	fr.Total += other.Total
 	fr.Missing += other.Missing
 	fr.Other += other.Other
-	if fr.Terms != nil && other.Terms != nil {
+	if other.Terms != nil {
+		if fr.Terms == nil {
+			fr.Terms = other.Terms
```

Comment:
So from what I understand, if there are no term facets, you can simply assign Other's term range facets and early exit.
If fr.numeric/date range facets are nil, why don't we do the same there?
just trying to understand.

Reply:
actually, nvm, I realised why.